fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Election.java

```diff
@@ -219,40 +238,32 @@ public static String generateEncryptedElectionVoteId(
    * @return the hash of all registered votes
    */
   public String computerRegisteredVotesHash() {
-    List<String> listOfVoteIds =
-        getElectionVersion() == ElectionVersion.OPEN_BALLOT
-            ? getListOfVoteIds(openVoteByPublicKey, ElectionVote::getId)
-            : getListOfVoteIds(encryptedVoteByPublicKey, ElectionEncryptedVote::getId);
-
-    if (listOfVoteIds.isEmpty()) {
+    String[] ids =
+        messageMap.values().stream()
+            .map(votesBySender::get)
+            // Merge lists and drop nulls
+            .flatMap(
+                electionVotes -> electionVotes != null ? electionVotes.stream() : Stream.empty())
```

Comment:
Do you have null-coalescing operators that would allow this ?

```suggestion
                electionVotes -> electionVotes?.stream() ?? Stream.empty())
```

Reply:
In Kotlin, yes
In java, I don't think so. At least not in java 8 to 11